src/Identifier/Ldap/ExtensionAdapter.php

```diff
@@ -161,6 +161,26 @@ public function unbind()
         $this->_connection = null;
     }
 
+    /**
+     * Search the LDAP directory
+     *
+     * @param string $baseDN Base DN for the directory
+     * @param string $filter search filter
+     * @return array
+     */
+    public function search($baseDN, $filter)
+    {
+        $this->_setErrorHandler();
+        $result = ldap_search($this->_connection, $baseDN, $filter);
+        $this->_unsetErrorHandler();
+
+        $this->_setErrorHandler();
+        $entries = ldap_get_entries($this->_connection, $result);
+        $this->_unsetErrorHandler();
```

Comment:
Should the error be thrown in the unsetErrorHandler() method? Right now if any of the ldap_ functions trigger an error they default error handler is not restored.

An alternative would be to replace both methods with a wrapper function like:

```php
$result = $this->convertErrorToException(function() use ($baseDN, $filter) {
   return ldap_search($this->_connection, $baseDN, $filter);
});
```
This wrapper could ensure that the error handler is restored before throwing an error.